scipy/spatial/transform/rotation.py

```diff
@@ -1461,6 +1518,103 @@ def magnitude(self):
         else:
             return angles
 
+    def reduce(self, other):
+        """Use this rotation to reduce the other.
```

Comment:
Hmm, my concept of "reduction" differs from what is currently implemented.

What I thought:
```
The reduction is done as follows:

1. For each rotation in `other` the closest rotation in `self` is found. 
   Let's denote them as ``p`` and ``q`` respectively.
2. The "reduced" rotation is computed as the rotation from the found closest rotation to 
   the given --- ``r = q.inv() * p``.
```
In your code no inversions are done, it seems strange to me. What do you think?

Also now the returned rotation contains `len(self)` rotations which also contradicts the Examples given (this is just a convention though).

Reply:
I agree with your concept of reduction.

> In your code no inversions are done, it seems strange to me.

Taking the inverse is correct.  I have updated the code.
The old code didn't use an inversion since it is not needed if you always use a symmetry group for reduction (a symmetry group contains the inverse of every element).

I had also forgotten to swap `p` and `q` when making `reduce` a method of the reducing class, rather than the class on which the reduction acts.